### DISCUSSION TITLE
How to handle non semver versions with more than 4 integers?

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#17723] [Request Help]
Recently, @adoptium has release a version "17.0.4.1+1". It looks like semver, but there's an extra digit to consider, since it's a fix on top of 17.0.4+11. Following the advice of #12063, I was using the build number as an extra number, but with the additional digit, I'm again stuck. How do I make sure that Renovate correctly compares the versions?

**Comment by PhilipAbed** [comment]
i dont think we support that out of the box
The default is semver check, 
if its failing at that you will need to change versioning for the problematic dependency using a package rules

**Reply by astellingwerf** [reply]
Hi @PhilipAbed, could you explain what you mean with:
> if its failing at that you will need to change versioning for the problematic dependency using a package rules

What would you configure in the package rule? I'm using a package rule with a regex versioning for the packages already, but there's just no way to safely compare 17.0.4+8 to 17.0.4.1+1 and future versions, and have it match to the correct update type.

If we'd wanna address that in a new version of Renovate, how would we go about that?

**Reply by PhilipAbed** [reply]
you are right, i dont think we can compare such versions, we will need special implementation i believe

**Reply by astellingwerf** [reply]
> we will need special implementation i believe

That's why I'm asking...

**Reply by PhilipAbed** [reply]
what versioning are you using? according to the code:
if you are using regex versioning: 
` const { major, minor, patch, build, prerelease, compatibility } = groups;`
you need to call the groups by these names
make your 4th group `build` in your regex i believe it will work

**Reply by PhilipAbed** [reply]
if i understand correctly:
for: `17.0.4.1+1`
this: `regex:^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)\\.(?<build>\\d+)\\+(?<prerelease>\\d+)$`

**Reply by viceice** [reply]
the `+1` doesn't mark a prerelease. it's build metadata. you can try nuget versioning

**Reply by astellingwerf** [reply]
In this case, the string also has the "jdk-" prefix, so I'm afraid that nuget is not gonna help me here.

The `RegExpVersioningApi` has a base `GenericVersioningApi` that does seem to do the right things when it comes to comparison, but the regex specific impl works too hard acting like it is sure that its value is semver compliant. That is fine if the values we're hoping to match is actually semver(y), but it works against us if that's not the case.
